classification/class_task.py

```diff
@@ -0,0 +1,91 @@
+
+import torch
+import numpy as np
+from torch.utils.tensorboard import SummaryWriter
+from tqdm import tqdm
+
+import sys
+sys.path.append('../')
+from logger_utils import *
+from networks import BayesianNetwork, MLP
+from config import DEVICE
+
+class BNN_Classification():
+    def __init__(self, label, parameters):
+        super().__init__()
+        self.writer = SummaryWriter(comment=f"_{label}_training")
+        self.label = label
+        self.lr = parameters['lr']
+        self.hidden_units = parameters['hidden_units']
+        self.mode = parameters['mode']
+        self.batch_size = parameters['batch_size']
+        self.num_batches = parameters['num_batches']
+        self.n_samples = parameters['train_samples']
+        self.test_samples = parameters['test_samples']
+        self.x_shape = parameters['x_shape']
+        self.classes = parameters['classes']
+        self.mu_init = parameters['mu_init']
+        self.rho_init = parameters['rho_init']
+        self.prior_init = parameters['prior_init']
+        self.mixture_prior = parameters['mixture_prior']
+        self.init_net()
+    
+    def init_net(self):
+        model_params = {
+            'input_shape': self.x_shape,
+            'classes': self.classes,
+            'batch_size': self.batch_size,
+            'hidden_units': self.hidden_units,
+            'mode': self.mode,
+            'mu_init': self.mu_init,
+            'rho_init': self.rho_init,
+            'prior_init': self.prior_init,
+            'mixture_prior': self.mixture_prior
+        }
+        self.net = BayesianNetwork(model_params).to(DEVICE)
+        self.optimiser = torch.optim.Adam(self.net.parameters(), lr=self.lr)
+        self.scheduler = torch.optim.lr_scheduler.StepLR(self.optimiser, step_size=5000, gamma=0.5)
```

Comment:
```suggestion
        self.scheduler = torch.optim.lr_scheduler.StepLR(self.optimiser, step_size=100, gamma=0.5)
```
If we want to use scheduler for classification, should decrease step_size since only training for 300 epochs.